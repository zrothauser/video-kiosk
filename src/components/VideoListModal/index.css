.b-video-index {
  position: absolute;
  height: 100%;
  width: 100vw;
  background: black;
  background-color: rgb(31, 30, 30);
  height: 100%;
  width: 100vw;
  display: flex;
  justify-content: center;
  position: absolute;
  z-index: 20;
}

.b-video-index__container {
  width: 80%;
  margin-top: 5%;
}

.b-video-index__table {
  width: 100%;
  font-weight: 700;
}

.b-video-index__cell {
  border-bottom: 1px solid rgb(46, 50, 57);
}

.b-video-index__cell:last-child {
  text-align: right;
}

.b-video-index__cell:nth-child(4) {
  text-align: center;
}

.b-video-index__head {
  font-size: 14px;
  letter-spacing: .16em;
}

.b-video-index__body {
  font-size: 20px;
  line-height: 2em;
}
