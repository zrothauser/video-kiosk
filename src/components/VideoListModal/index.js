// Dependencies
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { AllHtmlEntities as Entities } from 'html-entities';
import ScrollLock from 'react-scrolllock';
import Transition from 'react-transition-group/Transition';

// Helpers
import { convertSecondsToMinutesSeconds } from '../../utils/video';

// Styles
import './index.css';

// Transition
import * as transitions from '../transitions';

const {
  durations,
  mediumFadeStyles,
} = transitions;

// Set up helper object
const entities = new Entities();

class VideoListModal extends React.Component {
  renderRow(video) {
    return (
      <tr className="b-video-index__row" key={video.id}>
        <td className="b-video-index__cell">
          <Link
            to={`/video/${video.id}`}
            onClick={() => this.props.toggleVideoIndex()}
          >
            {entities.decode(video.title)}
          </Link>
        </td>
        <td className="b-video-index__cell">
          {video.parentCategoryTitle}
        </td>
        <td className="b-video-index__cell">
          {convertSecondsToMinutesSeconds(video.duration)}
        </td>
      </tr>
    );
  }

  render() {
    const {
      videos,
      visible,
    } = this.props;

    return (
      <Transition
        in={visible}
        timeout={durations.medium}
        appear
        mountOnEnter
        unmountOnExit
      >
        {state => (
          <div
            className="b-video-index"
            style={{
              ...mediumFadeStyles.default,
              ...mediumFadeStyles[state],
            }}
          >
            <div className="b-video-index__container">
              <table className="b-video-index__table">
                <thead className="b-video-index__head">
                  <tr>
                    <td className="b-video-index__cell">
                      TITLE
                    </td>
                    <td className="b-video-index__cell">
                      TOPIC
                    </td>
                    <td className="b-video-index__cell">
                      TIME
                    </td>
                  </tr>
                </thead>
                <tbody className="b-video-index__body">
                  {videos.map(video => this.renderRow(video))}
                </tbody>
              </table>
            </div>
            <ScrollLock />
          </div>
        )}
      </Transition>
    );
  }
}

VideoListModal.propTypes = {
  videos: PropTypes.arrayOf(PropTypes.object).isRequired,
  toggleVideoIndex: PropTypes.func.isRequired,
  visible: PropTypes.bool.isRequired,
};

export default VideoListModal;
