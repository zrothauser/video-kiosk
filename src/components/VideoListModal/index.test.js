import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import VideoListModal from './index';

describe('components', () => {
  describe('VideoListModal', () => {
    const mockProps = {
      videos: [],
      toggleVideoIndex: jest.fn(),
      visible: true,
    };

    it('renders without crashing', () => {
      shallow(<VideoListModal {...mockProps} />);
    });

    it('renders correctly', () => {
      const VideoListModalComponent = renderer.create(
        <Router>
          <VideoListModal {...mockProps} />
        </Router>,
      );

      expect(VideoListModalComponent).toMatchSnapshot();
    });
  });
});
