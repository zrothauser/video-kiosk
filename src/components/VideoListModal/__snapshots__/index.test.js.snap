// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components VideoListModal renders correctly 1`] = `
<div
  className="b-video-index"
  style={
    Object {
      "opacity": 0,
      "transition": "opacity 400ms ease-in-out",
    }
  }
>
  <div
    className="b-video-index__container"
  >
    <table
      className="b-video-index__table"
    >
      <thead
        className="b-video-index__head"
      >
        <tr>
          <td
            className="b-video-index__cell"
          >
            TITLE
          </td>
          <td
            className="b-video-index__cell"
          >
            TOPIC
          </td>
          <td
            className="b-video-index__cell"
          >
            TIME
          </td>
        </tr>
      </thead>
      <tbody
        className="b-video-index__body"
      />
    </table>
  </div>
</div>
`;
